jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Scanner.java

```diff
@@ -339,6 +338,14 @@ default void pathsChanged(Set<Path> paths) throws Exception
         void filesChanged(Set<String> filenames) throws Exception;
     }
 
+    /**
+     * Listener that notifies of the complete path change set.
+     */
+    public interface ChangeSetListener extends Listener
+    {
+        void pathsChanged(Map<Path, Notification> changeSet);
+    }
+
```

Comment:
Actually... @janbartel made the suggestion that instead of introducing a new listener, why not just have:
```java

    /**
     * Notification of files that changed in the last scan.
     */
    public interface BulkListener extends Listener
    {
        default void pathsChanged(Map<Path, Notification> changeSet) throws Exception
        {
            pathsChanged(changeSet.keySet());    
        }
        
        default void pathsChanged(Set<Path> paths) throws Exception
        {
            filesChanged(paths.stream().map(Path::toString).collect(Collectors.toSet()));
        }

        void filesChanged(Set<String> filenames) throws Exception;
    }
```

Deprecate the old methods if you like.